mteb/tasks/Retrieval/multilingual/XSimPlusPlusRetrieval.py

```diff
@@ -0,0 +1,298 @@
+    "ars_Arab",
+    "ceb_Latn",
+    "fuv_Latn",
+    "kac_Latn",
+    "lin_Latn",
+    "nno_Latn",
+    "sat_Olck",
+    "tel_Telu",
+    "ydd_Hebr",
+    "ary_Arab",
+    "ces_Latn",
+    "gaz_Latn",
+    "kam_Latn",
+    "lit_Latn",
+    "nob_Latn",
+    "scn_Latn",
+    "tgk_Cyrl",
+    "yor_Latn",
+    "arz_Arab",
+    "cjk_Latn",
+    "gla_Latn",
+    "kan_Knda",
+    "lmo_Latn",
+    "npi_Deva",
+    "shn_Mymr",
+    "tgl_Latn",
+    "yue_Hant",
+    "asm_Beng",
+    "ckb_Arab",
+    "gle_Latn",
+    "kas_Arab",
+    "ltg_Latn",
+    "nso_Latn",
+    "sin_Sinh",
+    "tha_Thai",
+    "zho_Hans",
+    "ast_Latn",
+    "crh_Latn",
+    "glg_Latn",
+    "kas_Deva",
+    "ltz_Latn",
+    "nus_Latn",
+    "slk_Latn",
+    "tir_Ethi",
+    "zho_Hant",
+    "awa_Deva",
+    "cym_Latn",
+    "grn_Latn",
+    "kat_Geor",
+    "lua_Latn",
+    "nya_Latn",
+    "slv_Latn",
+    "tpi_Latn",
+    "zsm_Latn",
+    "ayr_Latn",
+    "dan_Latn",
+    "guj_Gujr",
+    "kaz_Cyrl",
+    "lug_Latn",
+    "oci_Latn",
+    "smo_Latn",
+    "tsn_Latn",
+    "zul_Latn",
+    "azb_Arab",
+    "deu_Latn",
+    "hat_Latn",
+    "kbp_Latn",
+    "luo_Latn",
+    "ory_Orya",
+    "sna_Latn",
+    "tso_Latn",
+    "azj_Latn",
+    "dik_Latn",
+    "hau_Latn",
+    "kea_Latn",
+    "lus_Latn",
+    "pag_Latn",
+    "snd_Arab",
+    "tuk_Latn",
+    "bak_Cyrl",
+    "dyu_Latn",
+    "heb_Hebr",
+    "khk_Cyrl",
+    "lvs_Latn",
+    "pan_Guru",
+    "som_Latn",
+    "tum_Latn",
+]
+_SPLIT = ["dev", "devtest"]
+
+
+class XSimPlusPlusRetrieval(MultilingualTask, AbsTaskRetrieval):
+    metadata = TaskMetadata(
+        name="XSimPlusPlusRetrieval",
+        dataset={
+            "path": "jaygala24/xsimplusplus",
+            "revision": "07f92f877ea651659f3815884761c49191a1c80c",
+        },
+        description="xsim++ is dataset created to capture more subtle improvements in bitext mining by adding challenging negative examples.",
+        reference="https://arxiv.org/abs/2212.05409",
+        type="Retrieval",
+        category="s2s",
+        eval_splits=_SPLIT,
+        eval_langs={x: ["eng-Latn", x.replace("_", "-")] for x in _LANGUAGES},
+        main_score="ndcg_at_10",
+        date=("2022-08-01", "2022-12-20"),
+        form=["written"],
+        domains=["Web", "News", "Non-fiction"],
+        task_subtypes=[],
+        license="CC0",
+        socioeconomic_status="mixed",
+        annotations_creators="derived",
+        dialect=[],
+        text_creation="machine-translated and verified",
+        bibtex_citation="""@article{chen2023xsim,
+  title     = {xSIM++: An Improved Proxy to Bitext Mining Performance for Low-Resource Languages},
+  author    = {Mingda Chen and Kevin Heffernan and Onur cCelebi and Alexandre Mourachko and Holger Schwenk},
+  journal   = {Annual Meeting of the Association for Computational Linguistics},
+  year      = {2023},
+  doi       = {10.48550/arXiv.2306.12907},
+}""",
+        n_samples={"dev": 997, "devtest": 1012},
```

Comment:
Is there really only 1012 samples in total?

Any reason to use both the dev and devtest?

Reply:
The dataset has 1012 samples per language and there are 200 languages so in total it would amount to 200k samples.

The dataset has both dev and devtest splits but we can choose to only include the devtest as the eval split as people usually report the result for the same.

Reply:
We should remove the dev set (200k is already on the higher end). Please add the fast loading introduced in #572 to avoid extensive long loading times.

Reply:
Then feel free to add the points as well as updating the scores using the fast implementation (just to show that the implementation work as intended)